storage/remote/otlptranslator/prometheusremotewrite/metrics_to_prw_test.go

```diff
@@ -235,6 +237,356 @@ func TestFromMetrics(t *testing.T) {
+			expectedError: `invalid temporality and type combination for metric "test_metric_2"`,
+		},
+		{
+			name:       "cumulative histogram",
+			allowDelta: false,
+			inputSeries: []pmetric.Metric{
+				createOtelExpHistogram("test_histogram", pmetric.AggregationTemporalityCumulative, ts),
+			},
+			expectedSeries: []prompb.TimeSeries{
+				createPromNativeHistogram("test_histogram", prompb.Histogram_UNKNOWN, ts),
+			},
+		},
+		{
+			name:       "delta histogram when allowed",
+			allowDelta: true,
+			inputSeries: []pmetric.Metric{
+				createOtelExpHistogram("test_histogram_1", pmetric.AggregationTemporalityDelta, ts),
+				createOtelExpHistogram("test_histogram_2", pmetric.AggregationTemporalityCumulative, ts),
+			},
+			expectedSeries: []prompb.TimeSeries{
+				createPromNativeHistogram("test_histogram_1", prompb.Histogram_GAUGE, ts),
+				createPromNativeHistogram("test_histogram_2", prompb.Histogram_UNKNOWN, ts),
+			},
+		},
+		{
+			name:       "delta histogram when not allowed",
+			allowDelta: false,
+			inputSeries: []pmetric.Metric{
+				createOtelExpHistogram("test_histogram_1", pmetric.AggregationTemporalityDelta, ts),
+				createOtelExpHistogram("test_histogram_2", pmetric.AggregationTemporalityCumulative, ts),
+			},
+			expectedSeries: []prompb.TimeSeries{
+				createPromNativeHistogram("test_histogram_2", prompb.Histogram_UNKNOWN, ts),
+			},
+			expectedError: `invalid temporality and type combination for metric "test_histogram_1"`,
+		},
+		{
+			name:          "cumulative histogram with buckets",
+			allowDelta:    false,
+			convertToNHCB: true,
+			inputSeries: []pmetric.Metric{
+				createOtelExplicitHistogram("test_histogram", pmetric.AggregationTemporalityCumulative, ts),
+			},
+			expectedSeries: []prompb.TimeSeries{
+				createPromNHCB("test_histogram", prompb.Histogram_UNKNOWN, ts),
+			},
+		},
+		{
+			name:          "delta histogram with buckets when allowed",
+			allowDelta:    true,
+			convertToNHCB: true,
+			inputSeries: []pmetric.Metric{
+				createOtelExplicitHistogram("test_histogram_1", pmetric.AggregationTemporalityDelta, ts),
+				createOtelExplicitHistogram("test_histogram_2", pmetric.AggregationTemporalityCumulative, ts),
+			},
+			expectedSeries: []prompb.TimeSeries{
+				createPromNHCB("test_histogram_1", prompb.Histogram_GAUGE, ts),
+				createPromNHCB("test_histogram_2", prompb.Histogram_UNKNOWN, ts),
+			},
+		},
+		{
+			name:          "delta histogram with buckets when not allowed",
+			allowDelta:    false,
+			convertToNHCB: true,
+			inputSeries: []pmetric.Metric{
+				createOtelExpHistogram("test_histogram_1", pmetric.AggregationTemporalityDelta, ts),
+				createOtelExplicitHistogram("test_histogram_2", pmetric.AggregationTemporalityCumulative, ts),
+			},
+			expectedSeries: []prompb.TimeSeries{
+				createPromNativeHistogram("test_histogram_1", prompb.Histogram_GAUGE, ts),
+			},
+			expectedError: `invalid temporality and type combination for metric "test_histogram_1"`,
+		},
```

Comment:
Here `test_histogram_1` is delta, and `allowDelta` is false. Then when looking at `expectedSeries`  I see `test_histogram_1`. I expected the opposite, that when we reject deltas  we would only expected the cumulative value being ingested 🤔

Reply:
Same thing is repeating for the test cases below, I'm not sure if I'm missing something but the expectations here are strange to me

Reply:
No it was my bad - I messed up with copy-pasting between tests + the test was not checking for expected series if there was an error 🙃

As per the previous comment, it's been fixed in https://github.com/prometheus/prometheus/pull/16360/commits/0c25109c3772ab65876f69d1549044f5e64c2f7c 

I've also been mixing up explicit and exponential histograms in some test cases. Those have been fixed too and I've gone through the tests more thoroughly to check everything makes sense now.